helix-term/src/application.rs

```diff
@@ -245,6 +256,126 @@ impl Application {
+                    description,
+                    text,
+                    reason,
+                    all_threads_stopped,
+                    ..
+                }) => {
+                    debugger.is_running = false;
+                    let main = debugger
+                        .threads()
+                        .await
+                        .ok()
+                        .and_then(|threads| threads.get(0).cloned());
+
+                    if let Some(main) = main {
+                        let (bt, _) = debugger.stack_trace(main.id).await.unwrap();
+                        debugger.stack_pointer = bt.get(0).cloned();
+                        debugger.stopped_thread = Some(main.id);
+                    }
+
+                    let scope = match thread_id {
+                        Some(id) => format!("Thread {}", id),
+                        None => "Target".to_owned(),
+                    };
+
+                    let mut status = format!("{} stopped because of {}", scope, reason);
+                    if let Some(desc) = description {
+                        status.push_str(&format!(" {}", desc));
+                    }
+                    if let Some(text) = text {
+                        status.push_str(&format!(" {}", text));
+                    }
+                    if all_threads_stopped.unwrap_or_default() {
+                        status.push_str(" (all threads stopped)");
+                    }
+
+                    if let Some(helix_dap::StackFrame {
+                        source:
+                            Some(helix_dap::Source {
+                                path: Some(src), ..
+                            }),
+                        line,
+                        column,
+                        end_line,
+                        end_column,
+                        ..
+                    }) = &debugger.stack_pointer
+                    {
+                        let path = src.clone();
+                        let line = *line;
+                        let column = *column;
+                        let end_line = *end_line;
+                        let end_column = *end_column;
+                        self.editor
+                            .open(path, helix_view::editor::Action::Replace)
+                            .unwrap();
+
+                        let (view, doc) = current!(self.editor);
+                        let start = doc.text().line_to_char(line - 1) + column;
+                        if let Some(end_line) = end_line {
+                            let end =
+                                doc.text().line_to_char(end_line - 1) + end_column.unwrap_or(0);
+                            doc.set_selection(
+                                view.id,
+                                Selection::new(smallvec![Range::new(start, end)], 0),
+                            );
+                        } else {
+                            doc.set_selection(view.id, Selection::point(start));
```

Comment:
We need some extra check here. This simple goto behavior leads to editor crash when stepping in some unavailable code like `file:///rustc/0035d9dcecee49d1f7349932bfa52c05a6f83641/library/core/src/fmt/mod.rs` (Rust standard library). Probably we should display stop status in status bar and not show the empty buffer (especially don't try to move to desired position).

Reply:
Doesn't panic now, but this behavior needs to be considered.